### PR TITLE
feat(java): Add new weblog variant for native image

### PR DESCRIPTION
**Pull request by PerfectSlayer**
## Description

This PR adds a new weblog variant for `native-image`.
It's based on Spring 2.7 but build with native-image. For now, only tracing feature is supported from Java tracer so most of the tests like appsec are disabled.

## Check list

Your PR is not ready to be reviewed? Please save it as a draft :pray:

- [x] Follows the style guidelines of this project (See [how to easily lint the code](https://github.com/DataDog/system-tests/blob/main/docs/edit/lint.md))
- [ ] CI is passing
- [ ] There is at least one approval from code owners

Yes to all? Feel free to merge it whenever you want :heart:

### CODE REVIEW

**Review by smola — APPROVED** [review]
I wonder if we could make the app code itself just a build profile for the existing `spring-boot` application. But I guess it might not be trivial, so not blocking the PR for that.

**Review by smola — CHANGES_REQUESTED** [review]
You could add it to the CI matrix. See an example here: https://github.com/DataDog/system-tests/pull/734/files

**Comment by PerfectSlayer** [comment]
About reusing `spring-boot` weblog:

One of the main issue I had is `Spring native` is experimental in `2.7`. Most of features `spring-boot` weblog features does not compile with `native-image` (for example, I tried to add Cassandra client for the `integration` tests without success). Moreover, as only tracing is only supported by the tracer when using native-image, it might not make sense to reuse a fully featured app.

About CI matrix:

I will first see how the PR is received, then I would be glad to enable it by default in CI.
Basically, I am not very satisfied of all `@missing_features` I had to add 🤷 
But thanks for the commit info! 👍

**Review by robertomonteromiguel — CHANGES_REQUESTED** [review]
We need to talk about it: https://github.com/DataDog/system-tests/pull/752

**Comment by PerfectSlayer** [comment]
As discussed with @robertomonteromiguel this PR will be continued in #752 as he reused the original Spring app. 
You can scrap what you need from this PR (like maven configurations, docker build file, …).
